common.rkt

```diff
@@ -150,17 +145,14 @@
                     ((var? t)  (set! index (+ 1 index))
                                (state (extend-sub t (reified-index index) (state-sub st)) (state-diseq st) (state-types st)))
                     (else      st)))))
-    (cond
-      ((and (null? (state-diseq st)) (null? (state-types st))) (walk* tm results))
-      ((null? (state-types st)) (Ans 
-                                  (walk* tm results)
-                                  (list (walk* (cons '=/= (map pretty-diseq (state-diseq st))) results))))
-      ((null? (state-diseq st)) (Ans
-                                  (walk* tm results)
-                                  (walk* (map pretty-types (state-types st)) results)))
-      (else (Ans (walk* tm results) 
-                (append (walk* (map pretty-types (state-types st)) results) 
-                        (list (walk* (cons '=/= (map pretty-diseq (state-diseq st))) results) )))))))
+    (let* ((walked-sub (walk* tm results))
+          (diseq-null (null? (state-diseq st)))
+          (types-null (null? (state-types st)))
+          (types (walk* (map pretty-types (state-types st)) results))
+          (diseq (if diseq-null '() (list (walk* (cons '=/= (map pretty-diseq (state-diseq st))) results)))))
```

Comment:
To be more direct, it might be nice to give a name to `(map pretty-diseq (state-diseq st))`, and then be testing whether that is `null?` in this `if` condition.